__tests__/app.test.js

```diff
@@ -138,15 +138,146 @@ describe("GET /api/articles/:article_id", () => {
+    return request(app)
+      .get("/api/articles/6/comments")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.comments).toBeInstanceOf(Array);
+      });
+  });
+  it("responds with all comments, if there are comments", () => {
+    return request(app)
+      .get("/api/articles/1/comments")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.comments.length).toBe(11);
+      });
+  });
+  it("responds with status 200 and message for no comments", () => {
+    return request(app)
+      .get("/api/articles/2/comments")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.message).toBe("No comments");
+        expect(body.comments.length).toBe(0);
+      });
+  });
+  it("responds with status 200 and array contains comment objects with correct properties", () => {
+    return request(app)
+      .get("/api/articles/1/comments")
+      .expect(200)
+      .then(({ body }) => {
+        body.comments.forEach((comment) => {
+          expect(typeof comment.comment_id).toBe("number");
+          expect(typeof comment.votes).toBe("number");
+          expect(typeof comment.created_at).toBe("string");
+          expect(typeof comment.author).toBe("string");
+          expect(typeof comment.body).toBe("string");
+        });
+      });
+  });
+  it("responds with correct comment data", () => {
+    return request(app)
+      .get("/api/articles/6/comments")
+      .expect(200)
+      .then(({ body }) => {
+        const comment = body.comments[0];
+        expect(comment.body).toBe("This is a bad article name");
+        expect(comment.votes).toBe(1);
+        expect(comment.author).toBe("butter_bridge");
+      });
+  });
+  it("responds with status 400 for invalid non numeric id", () => {
+    return request(app)
+      .get("/api/articles/blah/comments")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.message).toBe("Invalid ID");
+      });
+  });
+  it("responds with status 404 if no article found with requested article_id", () => {
+    return request(app)
+      .get("/api/articles/5000/comments")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.message).toBe("Article not found");
+      });
+  });
+});
+
+describe("POST /api/articles/:article_id/comments", () => {
+  it("responds with status 201 and returns posted comment", () => {
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send({ username: "butter_bridge", body: "New comment" })
+      .expect(201)
+      .then(({ body }) => {
+        const expectedCommentResponse = body["comment added"];
+        expect(typeof expectedCommentResponse.comment_id).toBe("number");
+        expect(typeof expectedCommentResponse.votes).toBe("number");
+        expect(typeof expectedCommentResponse.created_at).toBe("string");
+        expect(typeof expectedCommentResponse.author).toBe("string");
+        expect(expectedCommentResponse.body).toBe("New comment");
+      });
+  });

```

Comment:
This is good, but you could also have a test, which ignores extra properties that come along side username and body, so long as at the very least the username and body is given this should be a successful request.